src/client/core/utils/dom.js

```diff
@@ -364,28 +367,81 @@ export function getElementDescription (el) {
     return res.join('');
 }
 
-export function getNextFocusableElement (element, reverse) {
-    const offset       = reverse ? -1 : 1;
-    let allFocusable = sortElementsByFocusingIndex(getFocusableElements(findDocument(element)));
+export function focusNextElement (element, reverse, skipRadioGroups) {
+    return new Promise(resolve => {
+        const nextElement = getNextFocusableElement(element, reverse, skipRadioGroups);
+
+        if (!nextElement)
+            resolve();
 
-    //NOTE: in all browsers except Mozilla and Opera focus sets on one radio set from group only.
-    // in Mozilla and Opera focus sets on any radio set.
-    if (isInputElement(element) && element.type === 'radio' && element.name !== '' && !browserUtils.isFirefox) {
-        allFocusable = arrayUtils.filter(allFocusable, item => {
-            return !item.name || item === element || item.name !== element.name;
+        focusBlurSandbox.focus(nextElement, () => {
+            resolve(nextElement);
         });
-    }
+    });
+}
+
+function filterFocusableElements (elements, sourceElement, skipRadioGroups) {
```

Comment:
What if we rename this method `filterRadioButtonElements` and will call it
```js
//before
allFocusable = filterFocusableElements(allFocusable, element, skipRadioGroups); 

//after
if (isRadioButtonElement(element))
    allFocusable = filterRadioButtonElements(allFocusable, element, skipRadioGroups); 
```
(not necessary)

Reply:
`fukterRadioButtonElements` is not correct name. If we have focused radio button whilh is last in radio group, the next focusable item could be any input in case if we press `tab`